megamek/src/megamek/client/ui/swing/CommonMenuBar.java

```diff
@@ -399,7 +397,13 @@ public void actionPerformed(ActionEvent event) {
         }
 
         // Pass the action on to each of our listeners.
-        actionListeners.forEach(l -> l.actionPerformed(event));
+        // This is a source of ConcurrentModificationException errors if dialogs are open during events
+        // but closed after handling them;
+        try {
```

Comment:
> Catching the ConcurrentMod will stop the iteration. Very generally speaking, that's probably not what one would like here. Could the iteration be done using an iterator? It looks old-style but it does not mind the modification in the first place and could be the better solution.

I'll give it a shot.  

My read was that the event we're catching, based on the timing, is the loading of the reinforcements; none of the windows falling after the MUL reinforcement dialog in the list of listeners (having been spawned after that dialog) should really care about this.  The Round Report doesn't care about events relating to force spawning as it will be populated over the course of the next round anyhow.  All other UI elements should come before the dialog we are closing (and deleting from our list, thus causing this ConcurrentMod) as we appear to keep them around and just adjust their visibility.

The real issue, to my mind, is that we're doing our own tracking of listening elements at all.  Shouldn't this be automatically handled somewhere?

Reply:
It looks like the event that's firing and causing the CME is just the completion of the user clicking on "fileUnitsReinforceRAT" or whatever in the menu bar, and this is _after_ the entire dialog has been closed and units have been loaded (using the example from CasualJoker's stream).

From my testing, catching the CME here is fine, and making the entire GUI event handling system do the right thing would take an immense amount of extra work.